File: 2024_java_basics/12-while-loops-lab/tasks/04-walking.java
import java.util.Scanner;

public class Walking {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int steps = 0;

        while (steps < 10000) {
            String input = scanner.nextLine();

            if (input.equals("Going home")) {
                int stepsHome = Integer.parseInt(scanner.nextLine());
                steps += stepsHome;
                break;
            } else {
                steps += Integer.parseInt(input);
            }
        }

        if (steps >= 10000) {
            int diff = steps - 10000;
            System.out.println("Goal reached! Good job!");
            System.out.println(diff + " steps over the goal!");
        } else {
            int diff = 10000 - steps;
            System.out.println(diff + " more steps to reach goal.");
        }

        scanner.close();
    }
}

File: 2022_c#_basics/12-while-loop-exercise/04-walking.cs
using System;

namespace _04_Walking
{
    internal class Program
    {
        static void Main(string[] args)
        {
            int steps = 0;

            while (steps < 10000)
            {
                string input = Console.ReadLine();

                if (input == "Going home")
                {
                    int stepsHome = int.Parse(Console.ReadLine());
                    steps += stepsHome;
                    break;
                }
                else
                {
                    steps += int.Parse(input);
                }
            }

            if (steps >= 10000)
            {
                int dif = steps - 10000;
                Console.WriteLine($"Goal reached! Good job!");
                Console.WriteLine($"{dif} steps over the goal!");
            }
            else
            {
                int dif = 10000 - steps;
                Console.WriteLine($"{dif} more steps to reach goal.");
            }
        }
    }
}
